src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h

```diff
@@ -54,17 +58,19 @@ class DataTransfer
         pumpMaxE,
     };
 
-    Matrix<double, double> maxPower;
-
-    bool LoadFromFolder(Study& study, const AnyString& folder, Area& area);
-    bool AutoTransferHours(Study& study, const AnyString& folder, Area& area);
-    void AutoTransferPower(Matrix<double, int32_t>& matrix,
-                                       const Matrix<double>::ColumnType& maxPower);
-    bool SupportForOldStudies(Study& study, const AnyString& folder, Area& area);
-
-
+private:
+    bool LoadDailyMaxPowersAndEnergies(const AnyString& folder, const Area& area);
+    bool SaveDailyMaxEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveMaxGenerationEnergy(const AnyString& folder, Area& area);
+    bool SaveMaxPumpingEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxGenPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPumpPowerAsHourly(const AnyString& folder, Area& area);
```

Comment:
Add a comment about legacy ? For versions newer than XXX, use YYY
For older versions, use ZZZ

Reply:
All of these functions are used when loading old study! HydroMaxTimeSeriesReader class is just for supporting old studies.